test/rbc/rbc_test.go

```diff
@@ -0,0 +1,250 @@
+package rbc
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+	"strconv"
+	"sync"
+	"testing"
+
+	"github.com/DE-labtory/cleisthenes"
+	"github.com/DE-labtory/cleisthenes/pb"
+	"github.com/DE-labtory/cleisthenes/rbc"
+	"github.com/DE-labtory/iLogger"
+)
+
+type mockHandler struct {
+	ServeRequestFunc func(msg cleisthenes.Message)
+}
+
+func newMockHandler() *mockHandler {
+	return &mockHandler{}
+}
+
+func (h *mockHandler) ServeRequest(msg cleisthenes.Message) {
+	h.ServeRequestFunc(msg)
+}
+
+type mockNode struct {
+	n, f int
+
+	// node's address
+	address cleisthenes.Address
+
+	rbcMap map[cleisthenes.Address]*rbc.RBC
+
+	server *cleisthenes.GrpcServer
+
+	connPool *cleisthenes.ConnectionPool
+
+	memberMap *cleisthenes.MemberMap
+}
+
+func newMockNode(n, f int, addr cleisthenes.Address) *mockNode {
+	server := cleisthenes.NewServer(addr)
+
+	connPool := cleisthenes.NewConnectionPool()
+
+	return &mockNode{
+		n:        n,
+		f:        f,
+		rbcMap:   make(map[cleisthenes.Address]*rbc.RBC, 0),
+		address:  addr,
+		server:   server,
+		connPool: connPool,
+	}
+}
+
+func (m *mockNode) setUpServer() {
+	handler := newMockHandler()
+	handler.ServeRequestFunc = func(msg cleisthenes.Message) {
+		m.serveRequestFunc(msg)
+	}
+
+	onConnection := func(conn cleisthenes.Connection) {
+		fmt.Println("[server] on connection")
+		conn.Handle(handler)
+		if err := conn.Start(); err != nil {
+			conn.Close()
+		}
+	}
+
+	m.server.OnConn(onConnection)
+}
+
+func (m *mockNode) listen() {
+	m.server.Listen()
+}
+
+func (m *mockNode) serveRequestFunc(msg cleisthenes.Message) {
+	host, port, _ := net.SplitHostPort(msg.GetRbc().Proposer)
+	iPort, _ := strconv.Atoi(port)
+	proposerAddr := cleisthenes.Address{
+		Ip:   host,
+		Port: uint16(iPort),
+	}
+	proposer := m.memberMap.Member(proposerAddr)
+
+	host, port, _ = net.SplitHostPort(msg.Sender)
+	iPort, _ = strconv.Atoi(port)
+	senderAddr := cleisthenes.Address{
+		Ip:   host,
+		Port: uint16(iPort),
+	}
+	sender := m.memberMap.Member(senderAddr)
+
+	r := m.rbcMap[proposer.Address]
+	if err := r.HandleMessage(sender, &pb.Message_Rbc{
+		Rbc: msg.GetRbc(),
+	}); err != nil {
+		fmt.Errorf("error in Handlemessage : %s", err.Error())
+	}
+}
+
+func setMockNodeList(t *testing.T, n, f int) []*mockNode {
+	var ip string = "127.0.0.1"
+	var port uint16 = 8000
+
+	nodeList := make([]*mockNode, 0)
+
+	for i := 0; i < n; i++ {
+		address := cleisthenes.Address{
+			Ip:   ip,
+			Port: GetAvailablePort(port),
+		}
+		node := newMockNode(n, f, address)
+		node.setUpServer()
+		go node.listen()
+
+		nodeList = append(nodeList, node)
+
+		port++
```

Comment:
no need logic

Reply:
we don't know when `listen` really listen. so i can't use `channel` to check `listen` logic will end. so i explicitly added `port++`. this can prevent duplicate port listen(Unless the other tests are running asynchronously!)

In fact, the test fails if you subtract that logic.